src/ObjCRuntime/Registrar.cs

```diff
@@ -235,8 +235,9 @@ public void Add (ObjCField field, ref List<Exception> exceptions)
 			{
 				// Check if there are any base types with the same property.
 				var base_type = BaseType;
+				var fieldNameInDictionary = (field.IsStatic ? "+" : "-") + field.Name;
```

Comment:
That seems to have the potential to create a lot of strings. Could we (later) try to:
- lookup different static/instance collections ? or
- only prepend "+" for static (and leave the instance without a sign) ?

Reply:
I chose the implementation that I thought would have the lowest minimum memory requirements.

* Having two different dictionaries, one for static and one for instance, would use more memory since there would be two dictionaries instead of one.
* The strings are transient, the GC will collect them, and as such they don't contribute to the minimum memory requirements.
* I could use a different key type (`Dictionary<Tuple<string, bool>, ObjCMember` for instance), but this means AOT-compiling a Dictionary with a struct as a key, which would increase code size (unless we already AOT-compile such a dictionary for something else).

I considered prepending `+` for static methods only, but then I thought of this (admittedly pathological) case:

```
[Export ("+foo")]
public void Foo () {}
```

which won't work with the static registrar, but works with the dynamic registrar (which is a use-case for Xamarin.Mac).

Finally I came to the conclusion I might be able to remove the whole problem when I make the static registrar do more at build time (see bug #[56022](https://bugzilla.xamarin.com/show_bug.cgi?id=56022)).